### ISSUE TITLE
Xamarin Forms: Issue with MediaElement audio playing

### ISSUE THREAD

**Issue by sreejithsree139** [xamarin/XamarinCommunityToolkit#684]
I am using `MediaElement` for playing audio. I have a list of items and I need to play audio one after another. When playing audio I will change the background color of that list item. Also If I tap an item, the audio of that selected item starts playing, also the color of that item is modifying. These features are working fine in android and ios. I have added a video of the android [here][1].

Because of [Native Crash Reporting][2] of `MediaElement`, I have added a [custom renderer][3] in the ios platform. After adding the custom renderer, there are some issues while playing audio in ios. First, it is playing the first item audio, then moving to 3rd one by changing the background color of 2nd and 3rd item. Then playing the 5th one and changing the color of the 4th and 5th items. I need to play items one by one. I have added a video of ios [here](https://drive.google.com/file/d/1WT-5RGt3-X_NMtd-iNDh-WsTxoyC3R2R/view).

My code:

	//Xaml
	<MediaElement 
		x:Name="audio_mediaelement"
		MediaEnded="AudioEnded"
		ShowsPlaybackControls="True">
		<MediaElement.HeightRequest>
			<OnIdiom x:TypeArguments="x:Double">
				<OnIdiom.Phone>80</OnIdiom.Phone>
				<OnIdiom.Tablet>120</OnIdiom.Tablet>
				<OnIdiom.Desktop>80</OnIdiom.Desktop>
			</OnIdiom>
		</MediaElement.HeightRequest>
	</MediaElement>

	//Initial audio source set up
	audio_mediaelement.Source = aduiourl

	//When audio end 
	public void AudioEnded(object sender, EventArgs args)
	{
		audioOrder = audioOrder + 1;
		if(audioOrder < listVerses.Count)
		{
			audio_mediaelement.Source = aduioFormat + listVerses[audioOrder].audioUrl;
			ChapterList.ItemsSource = listVerses;
			listVerses[audioOrder].BGColor = Color.FromHex("#f2ee71");
			ChapterList.ScrollTo(((IList)ChapterList.ItemsSource)[audioOrder-1], ScrollToPosition.Start, true);
		}
	}

I have uploaded a sample project [here][4] for reference. 

If I didn't add the MediaElement custom renderer, the audios play one after another(no issue in order). But at that time the app will hang when moving back to the previous page after playing audio.

VS in mac version: 8.8.3 (build 16)
Xcode: 12.3(12C33)
IOS version is 14.3

The sample having the navigation issue [here](https://drive.google.com/file/d/1z_KAqOnyxqYAzvTSyrh-g4vaSiNIoTjN/view?usp=sharing).
In this sample no issue with the audio playing order. The issue is with the navigation.
Page 1 -> page 2 -> MainPage. Then audio will play in the order without any issue. Then go back to the previous page using the previous option. Then again click the previous option on page 2. Then we can see the navigation issue there.
Native Crash Reporting issue occurring when we remove the media element custom renderer in ios.

  [1]: https://drive.google.com/file/d/14iGg3Ubs2zbcFPLjvfPssFNH1vbC6PkN/view?usp=sharing
  [2]: https://stackoverflow.com/questions/63762912/xamarin-forms-native-crash-reporting-in-ios-platform-due-to-mediaelement-featur
  [3]: https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Platform.iOS/Renderers/MediaElementRenderer.cs
  [4]: https://drive.google.com/file/d/1qoDjyeGyORC40yY4HGaj57hEfSkQ3FYN/view?usp=sharing

**Comment by AndreiMisiukevich** [comment]
Hi, can you reproduce the native crash on the latest pre-release XamarinCommunityToolkit version?

Also, you take a renderer from the wrong place (from Xamarin.Forms) when the MediaElement is in XamarinCommunityToolkit

https://github.com/xamarin/XamarinCommunityToolkit/blob/main/src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/iOS/MediaElementRenderer.ios.cs

**Comment by AndreiMisiukevich** [comment]
Well, as I see you try to use MediaElement from the wrong places (From Xamarin.Forms)

You should

1) Install the latest pre-release version of XCT https://www.nuget.org/packages/Xamarin.CommunityToolkit/
2) Use MediaElement from this nuget ```xct:MediaElement```

**Comment by AndreiMisiukevich** [comment]
CC @jfversluis

**Comment by jfversluis** [comment]
Hi there @sreejithsree139 thanks for the report!

As already pointed out by Andrei, the MediaElement moved from Forms to the Toolkit. This is a bit confusing, I totally get that. Could you try installing the Xamarin.CommunityToolkit package on your project and see if you can make it work with that?

If not, please open another issue, reference this one and we'll be happy to look at it. Thanks!

**Comment by sreejithsree139** [comment]
@jfversluis and @AndreiMisiukevich

I have installed Xamarin.CommunityToolkit(**1.0.0-pre6**) and updated xamarin forms version to **5.0.0.1791-pre5**. 

After that, I need to update the target framework version to Android 10.0(Q) and downloaded the 10.0 SDK. Since it is showing as an error.

In android, the audio playing is working fine like previous but in ios no changes in the issue. I tried to add the `MediaElementRenderer` but it has a lot of errors.

Sample without `MediaElementRenderer` is [here](https://drive.google.com/file/d/13dYEOQ6l99VtVlhx1bXnyEHT0SzfZYqX/view?usp=sharing).
Sample with `MediaElementRenderer` is [here](https://drive.google.com/file/d/1Bg2lBHBlythjZslHJFi_iT_BSNURR7zv/view?usp=sharing).

Sorry for the late response since I lost too much time on Android SDK installation.
